utils/ingress/ingress.go

```diff
@@ -0,0 +1,25 @@
+package ingress
+
+import (
+	"fmt"
+
+	"github.com/argoproj/argo-rollouts/pkg/apis/rollouts/v1alpha1"
+)
+
+// GetRolloutIngressKeys returns ingresses keys (namespace/ingressName) which are referenced by specified rollout
+func GetRolloutIngressKeys(rollout *v1alpha1.Rollout) []string {
+	ingressesSet := make(map[string]bool)
```

Comment:
why do we need this map? Couldn't we just append to the slice directly?

Reply:
It's a quick and easy way to de-dupe in case multiple `Rollouts` reference the same `StableIngress`. On second thought, I wonder whether this should be an error condition and whether we should add a check to ensure no two Rollouts reference the same Ingress.

Reply:
I'm not clear how you'd ever catch a dupe because it seems like you're adding `$x` and `$x-canary` to the map

Reply:
I see now how what I said yesterday was plain wrong - fixed :)